action/store.go

```diff
@@ -131,11 +142,89 @@ func init() {
 	storeCmd.Flags().StringVarP(&storeFlags.sha, "sha", "z", "", "git sha of test run")
 	storeCmd.Flags().StringVarP(&storeFlags.start, "time", "m", "", "time of test run")
 	storeCmd.Flags().StringVarP(&storeFlags.src, "src", "s", "", "source test output file")
+	storeCmd.MarkFlagFilename("src")
 	storeCmd.Flags().BoolVar(&storeFlags.bench, "bench", false, "indicate if test ran benchmarks")
 	storeCmd.Flags().BoolVar(&storeFlags.short, "short", false, "indicate if test is run with -short flag")
 	storeCmd.Flags().BoolVar(&storeFlags.race, "race", false, "indicate if test is run with -race flag")
 	storeCmd.Flags().StringSliceVar(&storeFlags.tags, "tags", []string{}, "comma-separated tags enabled for the run")
 	storeCmd.Flags().StringSliceVarP(&storeFlags.retests, "rerun", "r", []string{}, "comma-separated source output for retests")
 
 	RootCmd.AddCommand(storeCmd)
+	RegisterStorer(&SQLStorer{})
+}
+
+var (
+	registeredStorerNames []string
+	registeredStorers     = map[string]Storer{}
+)
+
+// RegisterStorer registers a storer with the `store` command.
+func RegisterStorer(storer Storer) {
+	fs := storer.FlagSet()
+	// add the storer name as a prefix to the flag names
+	fs.VisitAll(func(f *pflag.Flag) {
+		f.Name = fmt.Sprintf("%s.%s", storer.Name(), f.Name)
+		// clear shorthands to avoid conflicts with other flags
+		f.Shorthand = ""
+	})
+
+	registeredStorers[storer.Name()] = storer
+	registeredStorerNames = append(registeredStorerNames, storer.Name())
+
+	// add the flags to the store command
+	storeCmd.Flags().AddFlagSet(&fs)
+	storeCmd.Flags().Lookup("storer").Usage = fmt.Sprintf("storer name. supported options: %v", registeredStorerNames)
+}
+
+// Storer allows the `store` CLI to support different storers.
+type Storer interface {
+	// Name returns the storer's name.
+	Name() string
+	// FlagSet returns a flagset to be added to the command so that users can pass configuration options as flags.
+	FlagSet() (flagSet pflag.FlagSet)
+	// Required returns a list of required flags.
+	Required() []string
+	// Storer creates the storer instance.
+	Storer() (storer.Storer, error)
+}
+
+type SQLStorer struct {
```

Comment:
It is probably worth calling this a `PSQLStorer` since it is not a generic sql abstraction